### PR TITLE
Update cosign-codeowners (and fix OWNERS.md)

### LINKED ISSUE #208: Trim cosign-codeowners list
**Action required** if your name is below and you'd like to keep access to Cosign.

Hey all,

We've managed to accumulate quite a few cosign-codeowners:

- @Dentrax 
- @asraa 
- @bobcallaway 
- @dekkagaijin
- @developer-guy 
- @dlorenc 
- @font 
- @hectorj2f 
- @loosebazooka 
- @luhring 
- @lukehinds 
- @mattmoor 
- @n3wscott 
- @priyawadhwa 
- @vaikas 
- @znewman01 

This makes it hard to make changes that we'd want every owner's consent for (for example, https://github.com/sigstore/cosign/pull/2519 ).

Let's tidy up. I'm going to remove anybody who no longer can prioritize maintaining Cosign. If you'd like to stay on as a Cosign codeowner, please 👍 this comment by this time next week, Monday Jan 16th. (If you happen to miss it after we clean up I'm happy to add you back.)

### PR DESCRIPTION
**Pull request by znewman01**
Fixes #208.

CC:
- TSC members who are still on this list: @priyawadhwa @bobcallaway
- Folks being removed: @dlorenc @font @lukehinds @n3wscott @vaikas @luhring @loosebazooka @asraa 

Just cleaning up, we'll add you back if you like.

### CODE REVIEW

**Comment by znewman01** [comment]
CC @cpanato

File: github-sync/github-data/users.yaml [hunk 1]
```diff
@@ -25,8 +25,6 @@ users:
         role: member
       - name: architecture-doc-team
         role: member
-      - name: cosign-codeowners
-        role: member
```
Comment (SantiagoTorres) [inline github-sync/github-data/users.yaml:29]: Not for everybody, this is @asraa but the diff is not smart enough.

File: github-sync/github-data/users.yaml [hunk 2]
```diff
@@ -304,8 +298,6 @@ users:
     teams:
       - name: codeowners-maven-sigstore
         role: member
-      - name: cosign-codeowners
-        role: member
```
Comment (SantiagoTorres) [inline github-sync/github-data/users.yaml:308]: this is @loosebazooka

File: github-sync/github-data/users.yaml [hunk 3]
```diff
@@ -478,8 +464,6 @@ users:
         role: maintainer
       - name: scaffolding-codeowners
         role: member
-      - name: cosign-codeowners
-        role: member
```
Comment (SantiagoTorres) [inline github-sync/github-data/users.yaml:482]: and @vaikas

**Review by SantiagoTorres — APPROVED** [review]

File: github-sync/github-data/users.yaml (hunk 1 above)
Comment (haydentherapper) [reply github-sync/github-data/users.yaml:29]: MAsra should remain as she is actively developing and reviewing PRs, especially around verification.

Comment (znewman01) [reply github-sync/github-data/users.yaml:29]: I double-checked with her:
Screenshot 2023-01-18 at 11 45 57 AM

**Review by priyawadhwa — APPROVED** [review]

**Comment by haydentherapper** [comment]
@znewman01 I am concerned with the remaining list of codeowners. Some of the folks being removed are active in reviews and merging, and some of the folks who have asked to stay are not. I think this could affect Cosign development velocity. Happy to chat more offline, but wanted to raise this here.

**Comment by znewman01** [comment]
> Some of the folks being removed are active in reviews and merging,
I don't want to force anybody into having merge rights. If you can convince these folks to say they still want merge access, I will remove them from this PR.

> and some of the folks who have asked to stay are not.
I'm really just looking to tidy here. If you'd like to start a more formal process to review access, I think that should be separate.

**Comment by znewman01** [comment]
> @znewman01 @priyawadhwa @haydentherapper i think we should create a md file or put somewhere an emeritus people and add all those people that was a codeowner, like we have in k8s
> 
> Maybe we have just an emeritus.md and that is ir. Wdyt?

That feels like a good start; if we need to make this more formal later we can: https://github.com/sigstore/cosign/pull/2641

**Pull request merged**